### PR TITLE
Support clearance similar to devise/warden

### PR DESCRIPTION
**Pull request by jrochkind**
Clearance is a fairly popular alternative. https://github.com/thoughtbot/clearance

### CODE REVIEW

**Comment by martin308** [comment]
Thanks for this @jrochkind! Taking a look now 👀

File: lib/bugsnag/middleware/clearance_user.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      if notification.request_data[:rack_env] &&
+            notification.request_data[:rack_env]["clearance"] &&
+            notification.request_data[:rack_env]["clearance"].signed_in? &&
+            notification.request_data[:rack_env]["clearance"].current_user
+
+        # Extract useful user information
+        user = {}
+        user_object = notification.request_data[:rack_env]["clearance"].current_user
+        if user_object
+          # Build the user info for this scope
```
Comment (martin308) [inline lib/bugsnag/middleware/clearance_user.rb]: is this a valid comment for clearance? Does it have scopes like warden?

File: lib/bugsnag/middleware/clearance_user.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+          COMMON_USER_FIELDS.each do |field|
+            user[field] = user_object.send(field) if user_object.respond_to?(field)
+          end
+        end
+
+        # We merge the first warden scope down, so that it is the main "user" for the request
```
Comment (martin308) [inline lib/bugsnag/middleware/clearance_user.rb]: This comment also looks a little out of place for clearance

**Comment by martin308** [comment]
@jrochkind do we need to require `clearance_user.rb`? `warden_user.rb` is required in `bugsnag.rb`

File: lib/bugsnag/middleware/clearance_user.rb (hunk 1 above)
Comment (jrochkind) [inline lib/bugsnag/middleware/clearance_user.rb]: Clearance has no scopes, I'll remove the accidentally copy-pasted comment.

**Comment by jrochkind** [comment]
Removed/fixed the accidentally copy-pasted comments from the devise/warden version. 

Added the missing `require`. 

Thanks!

**Comment by kattrali** [comment]
Looks good to me 👍

**Comment by keeganlow** [comment]
@jrochkind, I just made a little tweak to this and merged - thanks!

**Pull request merged**